watcher/src/watcher.rs

```diff
@@ -194,7 +194,7 @@ impl Watcher {
                     }
 
                     Err(err) => {
-                        log::debug!(
+                        log::error!(
```

Comment:
I might be wrong, but wouldn't this be very noisy if everything is actually synced since the next block has not been published yet so we are getting 404s but they are expected?

Reply:
Good call out. I might be wrong, but I think this takes care of that by exiting the loop if the last recorded sync block is greater or equal to that of the ledger height: https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/watcher.rs#L122-L135

Reply:
I don't think thats what `max_block_height` is: https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/bin/main.rs#L151-L152
https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/config.rs#L30

Reply:
Noted, will double check

Reply:
Yes, you're absolutely correct, I'll find a solution

Reply:
Actually, this would only spam if watcher is started from the standalone binary, if started from the sync thread defined here: https://github.com/mobilecoinfoundation/mobilecoin/blob/master/watcher/src/watcher.rs#L330-L359 it establishes the limit of max_block_height to a max of the ledger height. Does watcher ever run in a standalone fashion?

Reply:
I think block explorer uses it directly...

Reply:
Added a simple bool flag to log errors if launched from mobilecoind. A more elaborate solution could be possible (if we need to determine watcher is behind when launched from the binary), keeping it simple is the best path.